how-to-guides/data-versioning/scripts/Version_datasets_in_model_training_runs.py

```diff
@@ -0,0 +1,110 @@
+    X_test, y_test = test[FEATURE_COLUMNS], test[TARGET_COLUMN]
+
+    rf = RandomForestClassifier(**params)
+    rf.fit(X_train, y_train)
+
+    score = rf.score(X_test, y_test)
+    return score
+
+
+#
+# Run model training and log dataset version, parameter and test score to Neptune
+#
+
+# Create a Neptune Run and start logging
+# run = neptune.init(project="common/quickstarts",
+#                    api_token="ANONYMOUS")
+run = neptune.init(project='common/data-versioning',
+                   api_token='ANONYMOUS')
+
+# Track dataset version
+run["datasets/train"].track_files(TRAIN_DATASET_PATH)
+run["datasets/test"].track_files(TEST_DATASET_PATH)
+
+# Log parameters
+PARAMS = {'n_estimators': 3,
+          'max_depth': 3,
+          'max_features': 1,
+          }
+run["parameters"] = PARAMS
+
+# Caclulate and log test score
+score = train_model(PARAMS, TRAIN_DATASET_PATH, TEST_DATASET_PATH)
+run["metrics/test_score"] = score
+
+# Get Neptune Run ID of the first, baseline model training run
+baseline_run_id = run['sys/id'].fetch()
+print(baseline_run_id)
+
+# Stop logging to the active Neptune Run
+run.stop()
+
+#
+# Change the model parameters
+# Run model training log dataset version, parameter and test score to Neptune
+#
+
+# Create a new Neptune Run and start logging
+# new_run = neptune.init(project="common/quickstarts",
+#                               api_token="ANONYMOUS",
+#                               run=baseline_run_id)
+
+new_run = neptune.init(project='common/data-versioning',
+                       api_token='ANONYMOUS')
+
+# Track dataset version
+new_run["datasets/train"].track_files(TRAIN_DATASET_PATH)
+new_run["datasets/test"].track_files(TEST_DATASET_PATH)
+
+# Query the baseline Neptune run
+# baseline_run = neptune.init(project="common/quickstarts", 
+#                               api_token="ANONYMOUS", 
+#                               run=baseline_run_id)
+baseline_run = neptune.init(project='common/data-versioning',
+                            api_token='ANONYMOUS',
+                            run=baseline_run_id,
+                            mode="read-only")
+
+# Fetch the dataset version of the baseline model training run
+baseline_run["datasets/train"].fetch_hash()
+
+# Check if dataset versions changed or not between the runs
+new_run.wait() # force asynchronous logging operations to finish
+assert baseline_run["datasets/train"].fetch_hash() == new_run[
+    "datasets/train"].fetch_hash()
+assert baseline_run["datasets/test"].fetch_hash() == new_run[
+    "datasets/test"].fetch_hash()
+
+# Select new parameters and log them to Neptune
+PARAMS = {'n_estimators': 3,
+          'max_depth': 2,
+          'max_features': 3,
+          }
+new_run["parameters"] = PARAMS
+
+# Calculate the test score and log it to Neptune
+score = train_model(PARAMS, TRAIN_DATASET_PATH, TEST_DATASET_PATH)
+new_run["metrics/test_score"] = score
+
+# Stop logging to the active Neptune Run
+new_run.stop()
```

Comment:
you probably want to stop baseline run as well

Reply:
Hmm, I thought that if the run is initialized in read-only mode I don't have to stop it.
Just conceptually though, never asked anyone about it.